elements/rocky-container-stackhpc/cleanup.d/80-cleanup-and-restore-repofiles

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+if [ ${DIB_DEBUG_TRACE:-0} -gt 0 ]; then
+    set -x
+fi
+set -eu
+set -o pipefail
+
+DIB_ROCKY_CONTAINER_STACKHPC_RESTORE_UPSTREAM_REPOFILES=${DIB_ROCKY_CONTAINER_STACKHPC_RESTORE_UPSTREAM_REPOFILES:-"false"}
+
+[ -n "$TARGET_ROOT" ]
+
+# Remove custom repos and disabled upstream repo files
+sudo rm -f ${TARGET_ROOT}/etc/yum.repos.d/*.repo
```

Comment:
Let's say I want to build a regular image without changing mirrors so I set ROCKY_USE_MIRRORS=false.
So my repo files never get copied to orig_repos
So unless I've missed something, this will delete the upstream repo files. Is that by design?

Reply:
I agree that this is a bit clunky. It was sort of by design as you don't have a DIB env var to control this (`ROCKY_USE_CUSTOM_DNF_MIRRORS` is a build arg passed in via `DIB_CONTAINERFILE_BUILDOPTS`) and my initial thought was that Kayobe reinstates the desired repo files early on in a `host configure` anyway. However, we do use this outside Kayobe, and in that case removing repo files whether you are using custom repos or not seems a bit rude.

Updated to only remove the contents of `/etc/yum.repos.d` if the `/tmp/orig_repos` directory exists, which is a reasonable proxy for using custom mirrors.